server/src/main/java/io/druid/client/BrokerServerView.java

```diff
@@ -188,6 +191,11 @@ private QueryableDruidServer removeServer(DruidServer server)
 
   private void serverAddedSegment(final DruidServerMetadata server, final DataSegment segment)
   {
+    if (segmentWatcherConfig.getWatchedTiers() != null && !segmentWatcherConfig.getWatchedTiers()
+                                                                               .contains(server.getTier())) {
```

Comment:
this may throw an NPE if `getTier()` returns `null`, depending on the implementation of the underlying Set. We should not rely on the actual implementation and check for null specifically